packages/neo-one-client-core/src/user/keystore/LedgerHandler.ts

```diff
@@ -0,0 +1,146 @@
+
+  public static readonly init = async () => {
+    const isSupported = await LedgerTransport.isSupported();
+    if (!isSupported) {
+      throw new LedgerNotSupportedError();
+    }
+    const paths = await LedgerTransport.list();
+    if (paths.length === 0) {
+      throw new LedgerNotDetectedError();
+    }
+    const ledger = await LedgerTransport.open(paths[0]);
+    ledger.setScrambleKey('NEO');
+
+    return new LedgerHandler(ledger);
+  };
+
+  private readonly ledger: HWLedger;
+  private readonly mutableQueue: LedgerQueueRequest[];
+  private mutableRunning: boolean;
+
+  private constructor(ledger: HWLedger) {
+    this.ledger = ledger;
+    this.mutableRunning = false;
+    this.mutableQueue = [];
+  }
+
+  public async getPublicKey(account = 0): Promise<PublicKeyString> {
+    const response = await this.send(Buffer.concat([keyRequestParams, Buffer.from('14', 'hex'), BIP44(account)]));
+
+    return crypto.toECPoint(response.message).toString('hex');
+  }
+  public async sign({ message, account }: { readonly message: Buffer; readonly account: number }): Promise<string> {
+    const data = Buffer.concat([message, BIP44(account)]);
+    const chunks = chunkBuffer(data, 250);
+    const initChunks = chunks.slice(0, chunks.length - 1);
+    const finalChunk = chunks[chunks.length - 1];
+
+    await Promise.all(
+      initChunks.map(async (chunk) => {
+        await this.send(Buffer.concat([initMessageParams, Buffer.from([chunk.length]), chunk]));
+      }),
+    );
+
+    const response = await this.send(
+      Buffer.concat([finalMessageParams, Buffer.from(finalChunk.length.toString(16), 'hex'), finalChunk]),
+    );
+
+    return response.message.toString('hex');
+  }
+
+  public async close(): Promise<void> {
+    await this.ledger.close();
+  }
+
+  private async send(msg: Buffer): Promise<LedgerResponse> {
+    // tslint:disable-next-line:promise-must-complete
+    const promise = new Promise<LedgerResponse>((resolve, reject) => {
+      this.mutableQueue.push({
+        msg,
+        resolve,
+        reject,
+      });
+    });
+
+    this.startSendLoop().catch(() => {
+      // do nothing
+    });
+
+    return promise;
+  }
+
+  private async startSendLoop(): Promise<void> {
+    if (this.mutableRunning) {
+      return;
+    }
+
+    this.mutableRunning = true;
+
+    let entry = this.mutableQueue.shift();
+    // tslint:disable-next-line:no-loop-statement
+    while (entry !== undefined) {
+      try {
+        const result = await this.doSend(entry.msg);
+        entry.resolve(result);
+      } catch (error) {
+        entry.reject(error);
+      }
+      entry = this.mutableQueue.shift();
+    }
+
+    this.mutableRunning = false;
+  }
+
+  private async doSend(msg: Buffer): Promise<LedgerResponse> {
+    const response = await this.ledger.exchange(msg);
+
+    return {
+      message: response.slice(0, response.length - 2),
+      status: response.slice(response.length - 2),
```

Comment:
Should throw an error if the status is not successful and instead of returning an object with `message` and `status`, just return the `message`.